storage/src/mmr/verification.rs

```diff
@@ -168,58 +237,6 @@ impl<H: CHasher> Proof<H> {
         Ok(peak_digests)
     }
 
-    /// Return the maximum size in bytes of any serialized `Proof`.
-    pub fn max_serialization_size() -> usize {
-        u64::SIZE + (u8::MAX as usize * H::Digest::SIZE)
-    }
-
-    /// Canonically serializes the `Proof` as:
-    /// ```text
-    ///    [0-8): size (u64 big-endian)
-    ///    [8-...): raw bytes of each hash, each of length `H::len()`
-    /// ```
-    pub fn serialize(&self) -> Vec<u8> {
-        // A proof should never contain more digests than the depth of the MMR, thus a single byte
-        // for encoding the length of the digests array still allows serializing MMRs up to 2^255
-        // elements.
-        assert!(
-            self.digests.len() <= u8::MAX as usize,
-            "too many digests in proof"
-        );
-
-        // Serialize the proof as a byte vector.
-        let bytes_len = u64::SIZE + (self.digests.len() * H::Digest::SIZE);
-        let mut bytes = Vec::with_capacity(bytes_len);
-        bytes.put_u64(self.size);
-        for hash in self.digests.iter() {
-            bytes.extend_from_slice(hash.as_ref());
-        }
-        assert_eq!(bytes.len(), bytes_len, "serialization length mismatch");
-        bytes.to_vec()
-    }
-
-    /// Deserializes a canonically encoded `Proof`. See `serialize` for the serialization format.
-    pub fn deserialize(bytes: &[u8]) -> Option<Self> {
-        let mut buf = bytes;
-        if buf.len() < u64::SIZE {
-            return None;
-        }
-        let size = buf.get_u64();
-
-        // A proof should divide neatly into the hash length and not contain more than 255 digests.
-        let buf_remaining = buf.remaining();
-        let digests_len = buf_remaining / H::Digest::SIZE;
```

Comment:
^^ This is how we were actually inferring length of the vector previously, with no explicit length encoding.